src/components/calcite-panel/calcite-panel.tsx

```diff
@@ -180,45 +362,130 @@ export class CalcitePanel {
       />
     ) : null;
 
-    const slotNode = <slot name={SLOTS.headerTrailingContent} />;
+    const slotNode = <slot name={SLOTS.headerActionsEnd} />;
+    const hasEndActions = getSlotted(el, SLOTS.headerActionsEnd);
 
-    return (
-      <div class={CSS.headerTrailingContent} key="header-trailing-content">
+    return hasEndActions || dismissibleNode ? (
+      <div
+        class={{ [CSS.headerActionsEnd]: true, [CSS.headerActions]: true }}
+        key="header-actions-end"
+      >
         {slotNode}
         {dismissibleNode}
       </div>
+    ) : null;
+  }
+
+  renderMenuItems(): VNode {
+    const { el, menuOpen, menuButtonEl } = this;
+
+    return (
+      <calcite-popover
```

Comment:
We'll use `calcite-dropdown` for this at some point right? Would we still place an action inside of the dropdown to trigger it?

Reply:
That's what I'm hoping for. I think we'd still use an Action for the trigger.